### DISCUSSION TITLE
Query ancestor nodes for match

### DISCUSSION THREAD

**Discussion by kentslaney** [godotengine/godot-proposals#10466] [Scripting]
Is there a way to implement something like the [`closest` method in JavaScript](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)? It could reduce fragile parent/child dependencies between nodes.

**Comment by Mickeon** [comment]
How "close" is [find_parent](https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-method-find-parent)?

**Reply by kentslaney** [reply]
Oh, thanks for the link. I tried searching for my description but nothing came up. I'll wait till I know the table of contents for the API docs before I assume anything else is missing. 

Individual pages for each method might be better for search engines than relying on section URL anchors, though LLMs might mitigate that somewhat. Google APIs are good about having unique pages for each function description.
